tests/platform_tests/mellanox/test_thermal_control.py

```diff
@@ -89,11 +81,31 @@ def test_set_psu_fan_speed(duthosts, rand_one_dut_hostname, mocker_factory):
 
     logging.info('Mock FAN presence...')
     single_fan_mocker.mock_presence()
-    assert wait_until(THERMAL_CONTROL_TEST_WAIT_TIME, THERMAL_CONTROL_TEST_CHECK_INTERVAL, check_cooling_cur_state, duthost, 10, operator.ne), \
-        'Current cooling state is {}'.format(get_cooling_cur_state(duthost))
-    logging.info('Wait {} seconds for the policy to take effect...'.format(THERMAL_CONTROL_TEST_CHECK_INTERVAL))
-    time.sleep(THERMAL_CONTROL_TEST_CHECK_INTERVAL)
+    wait_until(THERMAL_CONTROL_TEST_WAIT_TIME, THERMAL_CONTROL_TEST_CHECK_INTERVAL, check_cooling_cur_state, duthost, 10, operator.ne)
+    logging.info('Wait {} seconds for the policy to take effect...'.format(THERMAL_CONTROL_TEST_WAIT_TIME * 2))
+    # We have to wait THERMAL_CONTROL_TEST_WAIT_TIME * 2 seconds long here because:
+    #     1. We observe that on lionfish, it might take about 30 seconds for PSU to adjust
+    #     its fan speed to target value
+    #     2. Issue 2255767. There is chance that kernel might incorrectly change cooling state back to 10 after
+    #     user space thermal control adjust it to dynamic minimum value. This issue is set to "Wont Fix"
+    #     by hw-mgmt team. So we have to wait longer for the user space thermal control
+    #     to set fan speed to dynamic minimum value again (1 more minutes).
+    time.sleep(THERMAL_CONTROL_TEST_WAIT_TIME * 2)
     cooling_cur_state = get_cooling_cur_state(duthost)
+    if cooling_cur_state == 10:
```

Comment:
what does 10 means?

Reply:
It means max cooling level. I will change it to a constant.